app/grandchallenge/codebuild/tasks.py

```diff
@@ -0,0 +1,66 @@
+from celery import shared_task
+from django.apps import apps
+from django.conf import settings
+from django.db.transaction import on_commit
+
+from grandchallenge.algorithms.models import Algorithm
+from grandchallenge.codebuild.client import CodeBuildClient
+from grandchallenge.codebuild.models import Build
+
+
+@shared_task()
+def create_algorithm_image(*, pk):
+    GitHubWebhookMessage = apps.get_model(  # noqa: N806
+        app_label="github", model_name="GitHubWebhookMessage"
+    )
+    ghwm = GitHubWebhookMessage.objects.get(pk=pk)
+    if Build.objects.filter(project_name=ghwm.project_name).exists():
+        return
+    algorithm = Algorithm.objects.get(
+        repo_name=ghwm.payload["repository"]["full_name"]
+    )
+
+    client = CodeBuildClient(
+        project_name=ghwm.project_name, msg=ghwm, algorithm=algorithm
+    )
+    client.create_build_project(
+        source=f"{settings.PRIVATE_S3_STORAGE_KWARGS['bucket_name']}/{ghwm.zipfile.name}"
+    )
+    build_pk = client.start_build()
+    on_commit(
+        lambda: wait_for_build_completion.apply_async(
+            kwargs={"build_pk": str(build_pk)}
+        )
+    )
+
+
+@shared_task(bind=True, max_retries=100)
+def wait_for_build_completion(self, *, build_pk):
+    build = Build.objects.get(pk=build_pk)
+
+    client = CodeBuildClient(build_id=build.build_id)
+
+    status = client.get_build_status()
+    if status == "IN_PROGRESS":
+        self.retry(countdown=30)
```

Comment:
What happens when the time limit is exceeded? Can we limit that in code build?

Reply:
I think CodeBuild itself has a 60 minute timeout. I can either raise our own timeout to that, or I can check if that is configurable somewhere.